### ISSUE TITLE
How should we handle case by case declarations?

### ISSUE THREAD

**Issue by colmtuite** [colmtuite/bantamcss#8]
Sometimes in projects, you need to write custom CSS for a variety of reasons. You might need to use negative margins to nudge an element. You might need to correct cross browser rendering issues. The vast majority of the time, these hacks are not reusable at all, yet they are usually necessary. So, how should we handle them? Here are some options:

- Just use inline styles? I think this is the best option.
- Add a class to the element and style the classes in a separate file specifically for these hacks.

**Comment by enociz** [comment]
If I might revive this issue, I would like to suggest that inline styles might actually be the best option. 

Firstly, a class would contribute to bloat of CSS and it wouldn't have much use unless it was to be used again and again.

Secondly, inline styles were made for this use in the sense that they aren't reusable and also have a high level of specificity.

What other thoughts can others add?

**Comment by enociz** [comment]
However, classes are what is usually used for this.

**Comment by brianzelip** [comment]
Just found this repo and am really into it as I've been thinking about single purpose classes a lot lately and putting some of those thoughts into code.

As someone new and having only read the readme, I assumed `theme/` is where the styles being discussed in this issue would go. To me it's a project-specific customization thing. I've often either dumped these into one file, or multiple files of categorically-related css, and included them into the normal build process via `@import`, usually at the end of the import cascade.

**Comment by colmtuite** [comment]
Hey @brianzelip , glad you like it, looking forward to getting more input!

I've recently removed `/theme` but I didn't update the readme yet. I removed `/theme` because I'm planning to release a default BantamCSS theme, probably in a separate repo. Most projects will use their own local `/theme` folder.

Anyway, the `/theme` folder is intended to house reusable single-purpose classes. But this issue is about styles that are not reusable, e.g. arbitrary widths or heights like `width: 243px`.

There are various ways to handle these styles depending on how your project is set up. But I don't think it's smart to abstract these non-reusable styles into your CSS, because CSS is global and usually downloaded on every page. The best approach I've found so far is to set these styles inline.
